xbmc/video/VideoDatabase.cpp

```diff
@@ -11900,61 +11900,62 @@ void CVideoDatabase::GetVideoVersions(VideoDbContentType itemType,
   }
 }
 
-void CVideoDatabase::GetDefaultVideoVersion(VideoDbContentType itemType, int dbId, CFileItem& item)
+int CVideoDatabase::GetDefaultVideoVersionId(VideoDbContentType itemType, int dbId)
 {
   if (!m_pDB || !m_pDS)
-    return;
+    return -1;
 
   MediaType mediaType;
-  std::string strSQL;
+  std::string query;
 
   if (itemType == VideoDbContentType::MOVIES)
   {
     mediaType = MediaTypeMovie;
-    strSQL = PrepareSQL("SELECT videoversiontype.name AS name,"
-                        "  videoversiontype.id AS id,"
-                        "  videoversion.idFile AS idFile "
-                        "FROM videoversiontype"
-                        "  JOIN videoversion ON"
-                        "    videoversion.idType = videoversiontype.id"
-                        "  JOIN movie ON"
-                        "    movie.idFile = videoversion.idFile "
-                        "WHERE movie.idMovie = %i",
-                        dbId);
+    query = PrepareSQL("SELECT idFile FROM movie WHERE movie.idMovie = %i", dbId);
   }
   else
-    return;
+    return -1;
+
+  int id = -1;
 
   try
   {
-    m_pDS->query(strSQL);
-
-    if (!m_pDS->eof())
+    m_pDS->query(query);
+    if (m_pDS->num_rows() > 0)
     {
-      std::string name = m_pDS->fv("name").get_asString();
-      int id = m_pDS->fv("id").get_asInt();
-      int idFile = m_pDS->fv("idFile").get_asInt();
-
-      CVideoInfoTag infoTag;
-      if (GetFileInfo("", infoTag, idFile))
-      {
-        infoTag.m_type = MediaTypeVideoVersion;
-        infoTag.m_iDbId = idFile;
-        infoTag.m_idVideoVersion = id;
-        infoTag.m_typeVideoVersion = name;
-        infoTag.m_strTitle = name;
-
-        item.SetFromVideoInfoTag(infoTag);
-        item.m_strTitle = name;
-        item.SetLabel(name);
-      }
+      id = m_pDS->fv(0).get_asInt();
     }
-    m_pDS->close();
   }
   catch (...)
   {
     CLog::Log(LOGERROR, "{} failed for {} {}", __FUNCTION__, mediaType, dbId);
   }
+
+  return id;
+}
+
+void CVideoDatabase::GetDefaultVideoVersion(VideoDbContentType itemType, int dbId, CFileItem& item)
+{
```

Comment:
I could have missed something but looks to me like GetDetailsByTypeAndId() already does what this revised function is doing (return FileItem for a given item type and id)

Reply:
This one just picks the default item in result of GetVideoVersions. The GetVideoVersions has special handling for Video Versions file items. It's not in GetDetailsByTypeAndId.

Reply:
OK I see that it makes up the information of the VideoInfoTags and it looks like you're trying to consolidate with other existing code. Looking forward to the next steps.